### PR TITLE
Softmax with temperature + Wrap up functions + Small fixes

### PR DESCRIPTION
**Pull request by GrunCrow**
- Wrapped up functions run_predict and load_and_prepare_model so the script can be runned with only one function call.
- Created a custom softmax that will add a temperature parameter, so a softer or harder softmax can be applied to the logits
- Small fixes like:
            - update bounding box format saved to the json
            - json saved for each new processed prediction
            - image plot and saved if flag activated
            - output json with indent so it will be more "readable"

Old predictions:

[
    {
        "image_id": "40_20201024 (1).JPG",
        "category_id": 9,
        "bbox": [
            863.661865234375,
            1.06390380859375,
            481.6708984375,
            908.3948974609375
        ],
        "score": 0.9582054615020752,
        "activations": [
            4.18474792240886e-06,
            3.595745874918066e-05,
            3.976739662903128e-06,
            4.0920795072452165e-06,
            2.5987499611801468e-05,
            3.883285444317153e-06,
            7.198200182756409e-05,
            2.2586307295568986e-06,
            2.7678781407303177e-05,
            0.9582054615020752,
            5.205633442528779e-06,
            2.7677422167471377e-06,
            4.9444941396359354e-05,
            2.4394498723268043e-06,
            1.454145422030706e-05
        ]
    },

New predictions:
[
    {
        "image_id": "40_20201024 (1).JPG",
        "category_id": 1000,
        "bbox": [
            863.661865234375,
            1.06390380859375,
            481.6708984375,
            908.3948974609375
        ],
        "score": 1,
        "activations": [
            0.05009071156382561,
            0.06211129203438759,
            0.04983598366379738,
            0.049978673458099365,
            0.060126740485429764,
            0.049717605113983154,
            0.06657568365335464,
            0.04709499701857567,
            0.06050705164670944,
            0.2363872528076172,
            0.05119619518518448,
            0.04806210845708847,
            0.06412161886692047,
            0.04745909199118614,
            0.05673506483435631
        ],
      
(this logits has been created with a temperature of 10, which is too much for this case, it was just an example)

### CODE REVIEW

**Review by justinkay — APPROVED** [review]
Looks good! 
I'll change the `bbox_xywh` output from the prediction to be something like `bbox_xcycwh` -- NMS expects the x,y to be the center of the box 🙄
